### PR TITLE
[wip] Actions dependency

### PR DESCRIPTION
**Pull request by kmike**
It's also an option to add this dependency to web-poet, as ZyteApiActions.

The implementation is a bit tricky; this is the first time we have a parametrized dependency. It might require changes in andi (https://github.com/scrapinghub/andi/pull/24) and in other packages (scrapy-poet?). Only Python 3.9+ is supported here.

### CODE REVIEW

File: scrapy_zyte_api/page_inputs.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    urlMatchingOptions: Optional[str]
+    selector: Optional[_Selector]
+    onError: Optional[str]
+    delay: Optional[float]
+    timeout: Optional[float]
+    id: Optional[str]
+    keyword: Optional[str]
+    key: Optional[str]
+    source: Optional[str]
+    options: Optional[dict]
```
Comment (Gallaecio) [inline scrapy_zyte_api/page_inputs.py:25]: Wouldn’t it be better to have an approach that does not require defining action top-level fields here?

Comment (kmike) [reply scrapy_zyte_api/page_inputs.py:25]: This section is not required; the code would work fine without it as-is. If there are other top-level fields, I think it also should work as-is. I added it to have some type annotations for the users, via TypedDict. Would you prefer to have it not type checked?

Comment (Gallaecio) [reply scrapy_zyte_api/page_inputs.py:25]: I was not familiar with `TypedDict`, nice.

**Comment by wRAR** [comment]
Superseded by https://github.com/scrapy-plugins/scrapy-zyte-api/pull/181

**Pull request closed without merging**